packages/store/addon/-private/system/store/record-data-store-wrapper.ts

```diff
@@ -114,15 +123,32 @@ export default class RecordDataStoreWrapper implements IRecordDataStoreWrapper {
   inverseForRelationship(modelName: string, key: string): string {
     const modelClass = this._store.modelFor(modelName);
     const definition = upgradeForInternal(this.relationshipsDefinitionFor(modelName)[key]);
-
-    return definition._inverseKey(this._store, modelClass);
+    if (CUSTOM_MODEL_CLASS) {
+      if (definition.inverse !== undefined) {
+        return definition.inverse;
+      } else {
+        return definition._inverseKey(this._store, modelClass);
+      }
+    } else {
+      return definition._inverseKey(this._store, modelClass);
+    }
   }
 
   inverseIsAsyncForRelationship(modelName: string, key: string): boolean {
     const modelClass = this._store.modelFor(modelName);
     const definition = upgradeForInternal(this.relationshipsDefinitionFor(modelName)[key]);
-
-    return definition._inverseIsAsync(this._store, modelClass);
+    if (CUSTOM_MODEL_CLASS) {
+      if (definition.inverse === null) {
+        return true;
```

Comment:
why would this be true?

Reply:
we looked into the existing behavior and existing behavior is to return false when unknowable